boards/wio_terminal/src/display.rs

```diff
@@ -38,21 +41,97 @@ pub struct Display {
-pub type LCD = Ili9341<SPIInterface<LcdSpi, LcdDc, LcdCs>, LcdReset>;
+pub type LCD = Ili9341<SPIInterface<LcdDevice, LcdDc>, LcdReset>;
 
 pub use ili9341::Scroller;
 
+pub struct LcdSpiDevice<DC> {
+    spi: LcdSpi,
+    dc: DC,
+}
+
+impl<DC> LcdSpiDevice<DC> {
+    fn new(spi: LcdSpi, dc: DC) -> Self {
+        Self { spi, dc }
+    }
+}
+use atsamd_hal::ehal::spi::{ErrorType as SpiErrorType, Operation};
+impl<Cs> SpiErrorType for LcdSpiDevice<Cs> {
+    type Error = spi::Error;
+}
+impl<Cs> ehal::spi::SpiDevice<u8> for LcdSpiDevice<Cs> {
+    /// Perform a transaction against the device.
+    ///
+    /// - Locks the bus
+    /// - Asserts the CS (Chip Select) pin.
+    /// - Performs all the operations.
+    /// - [Flushes](SpiBus::flush) the bus.
+    /// - Deasserts the CS pin.
+    /// - Unlocks the bus.
+    ///
+    /// The locking mechanism is implementation-defined. The only requirement is it must prevent two
+    /// transactions from executing concurrently against the same bus. Examples of implementations are:
+    /// critical sections, blocking mutexes, returning an error or panicking if the bus is already busy.
+    ///
+    /// On bus errors the implementation should try to deassert CS.
+    /// If an error occurs while deasserting CS the bus error should take priority as the return value.
+    fn transaction(&mut self, operations: &mut [Operation<'_, u8>]) -> Result<(), Self::Error> {
+        todo!()
+    }
+
+    /// Do a read within a transaction.
+    ///
+    /// This is a convenience method equivalent to `device.transaction(&mut [Operation::Read(buf)])`.
+    ///
+    /// See also: [`SpiDevice::transaction`], [`SpiBus::read`]
+    #[inline]
+    fn read(&mut self, buf: &mut [u8]) -> Result<(), Self::Error> {
+        Ok(())
+    }
+
+    /// Do a write within a transaction.
+    ///
+    /// This is a convenience method equivalent to `device.transaction(&mut [Operation::Write(buf)])`.
+    ///
+    /// See also: [`SpiDevice::transaction`], [`SpiBus::write`]
+    #[inline]
+    fn write(&mut self, buf: &[u8]) -> Result<(), Self::Error> {
+        SpiBus::write(&mut self.spi, buf)
+    }
+
+    /// Do a transfer within a transaction.
+    ///
+    /// This is a convenience method equivalent to `device.transaction(&mut [Operation::Transfer(read, write)]`.
+    ///
+    /// See also: [`SpiDevice::transaction`], [`SpiBus::transfer`]
+    #[inline]
+    fn transfer(&mut self, read: &mut [u8], write: &[u8]) -> Result<(), Self::Error> {
+        SpiBus::write(&mut self.spi, write)
+    }
+
+    /// Do an in-place transfer within a transaction.
+    ///
+    /// This is a convenience method equivalent to `device.transaction(&mut [Operation::TransferInPlace(buf)]`.
+    ///
+    /// See also: [`SpiDevice::transaction`], [`SpiBus::transfer_in_place`]
+    #[inline]
+    fn transfer_in_place(&mut self, buf: &mut [u8]) -> Result<(), Self::Error> {
+        SpiBus::write(&mut self.spi, buf)
+    }
+}
+
```

Comment:
Here, I would advise looking at [`embedded_hal_bus`](https://docs.rs/embedded-hal-bus/latest/embedded_hal_bus/spi/index.html) rather than trying to implement `SpiDevice` yourself. I has nice utilities for connecting an `SpiBus` to an `SpiDevice`. In your case, if the bus won't be shared with other devices, you could use `ExclusiveDevice`.

Reply:
I will  consider it. Thank you for the advice.